srtcore/queue.cpp

```diff
@@ -1097,6 +1097,14 @@ void CRendezvousQueue::updateConnStatus(EReadStatus rst, EConnectStatus cst, con
     for (vector<LinkStatusInfo>::iterator i = ufailed.begin(); i != ufailed.end(); ++i)
     {
         HLOGC(cnlog.Debug, log << "updateConnStatus: COMPLETING dep objects update on failed @" << i->id);
+        // This is still done without locks, but a successful call to `locateSocket` should
+        // ensure that this entity will not be deleted in the next 1 second.
+        CUDTSocket* s = CUDT::uglobal()->locateSocket(i->id);
+        if (!s)
```

Comment:
```suggestion
        if (CUDT::uglobal()->locateSocket(i->id) == NULL)
```

Reply:
I'll retest it on Windows first. When this still proves to be a problem, we can't merge it, even as sanity check.